.github/workflows/continuous-integration.yml

```
@@ -37,14 +37,24 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip' # Enable pip-caching
```

Since we have the file `requirements_extra.txt` which is not automatically checked by `setup-python`, I suspect that we need to specify the `cache-dependency-path` as described in the documentation.

```suggestion
          cache: 'pip' # Enable pip-caching
          cache-dependency-path: |
            pyproject.toml
            requirements.txt
            requirements_extra.txt
```